x-pack/platform/plugins/shared/fleet/server/services/epm/packages/update_custom_integration.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type {
+  ElasticsearchClient,
+  SavedObject,
+  SavedObjectsClientContract,
+} from '@kbn/core/server';
+import { load, dump } from 'js-yaml';
+
+import {
+  PACKAGES_SAVED_OBJECT_TYPE,
+  PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  SO_SEARCH_LIMIT,
+} from '../../../../common/constants';
+
+import { packagePolicyService } from '../../package_policy';
+
+import { createArchiveIteratorFromMap } from '../archive/archive_iterator';
+
+import { appContextService } from '../../app_context';
+
+import type { PackageInstallContext } from '../../../../common/types';
+
+// Define a type for the integration attributes
+interface IntegrationAttributes {
+  version: string;
+  install_source: string;
+  [key: string]: any;
+}
+
+import { getInstalledPackageWithAssets } from './get';
+
+import { installPackageWithStateMachine } from './install';
+import { CustomIntegrationNotFoundError } from './custom_integrations/validation/check_custom_integration';
+
+export async function updateCustomIntegration(
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  id: string,
+  fields: {
+    readMeData?: string;
+    categories?: string[];
+  }
+) {
+  try {
+    // Get the current integration using the id
+    const integration: SavedObject<IntegrationAttributes> = await soClient.get(
+      PACKAGES_SAVED_OBJECT_TYPE,
+      id
+    );
+    if (!integration) {
+      throw new Error(`Integration with ID ${id} not found`);
```

Comment:
should be at least `FleetError` to be mapped to 4xx error response https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/errors/handlers.ts#L64